tests/CMakeLists.txt

```diff
@@ -61,20 +73,22 @@ add_test(unit_tests_size_3     ${TEST_WRAP} --vg unit_tests_size 3)
 add_test(unit_tests_size_2     ${TEST_WRAP} --vg unit_tests_size 2)
 add_test(unit_tests_size_1     ${TEST_WRAP} --vg unit_tests_size 1)
 add_test(unit_tests            ${TEST_WRAP} --vg unit_tests ${CMAKE_CURRENT_SOURCE_DIR}/threads4.conf)
-add_test(router_tests          ${TEST_WRAP} -s ${CMAKE_CURRENT_SOURCE_DIR}/router_engine_test.py -v)
-add_test(management_tests      ${TEST_WRAP} -m unittest -v management)
-
-macro(add_system_test test)
-  add_test(${test} ${TEST_WRAP} -m ${test} -v)
-endmacro(add_system_test)
 
-add_system_test(system_tests_broker)
-add_system_test(system_tests_management)
-add_system_test(system_tests_one_router)
-add_system_test(system_tests_qdmanage)
-add_system_test(system_tests_qdstat)
-add_system_test(system_tests_two_routers)
-add_system_test(system_tests_link_routes)
+# Add all sytem_tests* using add_test
+add_test(router_tests             ${TEST_WRAP} -s ${CMAKE_CURRENT_SOURCE_DIR}/router_engine_test.py -v)
+add_test(system_tests_broker      ${TEST_WRAP} -s ${CMAKE_CURRENT_BINARY_DIR}/system_tests_broker.py -v)
```

Comment:
Why do you move the tests to the build folder when you could simply run them from the source folder here?  Then you could run the "configured" tests from the build folder and the non-configured ones from the source folder.